test/jdk/java/lang/ClassLoader/loadLibraryUnload/LoadLibraryUnload.java

```diff
@@ -162,14 +162,14 @@ public static void main(String[] args) throws Exception {
         exceptions.clear();
         // Wait for the canary for each of the libraries to be GC'd
         // before exiting the test.
+        int dequeueCount = 0;
         for (int i = 0; i < LOADER_COUNT; i++) {
             System.gc();
-            var res = refQueue.remove(Utils.adjustTimeout(30 * 1000L));
-            System.out.println(i + " dequeued: " + res);
-            if (res == null) {
-                Asserts.fail("Too few cleared WeakReferences");
-            }
+            if (refQueue.remove(Utils.adjustTimeout(10 * 1000L)) != null)
+                dequeueCount++;
         }
+        Asserts.assertEquals(dequeueCount, LOADER_COUNT, "Too few cleared WeakReferences");
```

Comment:
This only invokes GC `LOADER_COUNT` times.  I think you may want something like what ForceGC does:

```
        int retries = (int)(Math.round(1000L * TIMEOUT_FACTOR) / 200);
        for (; retries >= 0; retries--) {
            if (dequeueCount == LOADER_COUNT) return;
            
            System.gc();
            refQueue.remove(200L);
        }
```

Reply:
I'll switch to using ForceGC.wait to wait for all the WeakReferences to be cleared.
Using existing logic for the gc and timing.